cannula/views.py

```diff
@@ -10100,3 +10103,35 @@ def grouped_data_generator(grouped_data):
     }
 
     return render(request, 'cannula/kp_{0}.html'.format(OrgUnit.get_level_field(org_unit_level)), context)
+
+#reports logic
+def indexreport(request):
+    context = {
+        'validation_rules': ValidationRule.objects.all().values_list('id', 'name')
+    }
+    return render(request, 'cannula/index_reports.html', context)
+
+def reports_sites_2016_to_2018(request):
+    context = {
+    'id' : 1,
+    } 
+    return render(request, 'cannula/performance_summary_oct_2016–sep_2017.html', context)
+
+def reports_sites_2017_to_2018(request):
+    context = {
+    'id' : 1,
+    } 
+    return render(request, 'cannula/performance_summary_oct_2017–sep_2018.html', context)
+
+def downloadreport(request, path):
+    file_path = os.path.join(settings.MEDIA_ROOT, path)
```

Comment:
Big security problem, unsanitised user input. Below is some code to illustrate the problem.
`
import os

good_path = os.path.join('/var/www/rhites_ec_www/static/media', 'report_2016.pdf')
print(good_path)
bad_path = os.path.join('/var/www/rhites_ec_www/static/media', '../../../../../etc/passwd')
print(bad_path)
print(os.path.realpath(bad_path))
`

Reply:
Dear Edward,
Thanks for the notes, just asking are you making the changes?

Reply:
Already done